### PR TITLE
buildah, bud: support --jobs=N for parallel execution

### PR DESCRIPTION
**Pull request by giuseppe**
/kind feature

#### What this PR does / why we need it:

it enables running multi stages Containerfiles in parallel

#### How to verify it

buildah bud --jobs=$(nproc) .....

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
add support for running multi stages Containerfiles in parallel
```

### CODE REVIEW

File: docs/buildah-bud.md [hunk 1]
```diff
@@ -278,6 +278,12 @@ container technology).
 Note: You can also override the default isolation type by setting the
 BUILDAH\_ISOLATION environment variable.  `export BUILDAH_ISOLATION=oci`
 
+**--jobs** *N*
+
+Run up to N concurrent stages in parallel.  If the number of jobs is greater than 1,
```
Comment (rhatdan) [inline docs/buildah-bud.md:283]: What is the default value?  1?

Comment (rhatdan) [reply docs/buildah-bud.md:283]: What does value 0  Mean?

Comment (giuseppe) [reply docs/buildah-bud.md:283]: the default value is 1.  0 is treated as 1 in the current implementation, should we raise an error instead?  make at least complains with `-j 0`:
```
$ make -j0 
make: the '-j' option requires a positive integer argument
```

Comment (rhatdan) [reply docs/buildah-bud.md:283]: Why not make 0 unlimited and >1 be the number of threads, and eliminate the negative numbers?

File: tests/bud.bats [hunk 2]
```diff
@@ -1384,6 +1384,9 @@ function _test_http() {
   target=busybox-derived
   run_buildah bud --signature-policy ${TESTSDIR}/policy.json -t ${target} -f ${TESTSDIR}/bud/copy-from/Dockerfile3 ${TESTSDIR}/bud/copy-from
   run_buildah bud --signature-policy ${TESTSDIR}/policy.json -t ${target} -f ${TESTSDIR}/bud/copy-from/Dockerfile4 ${TESTSDIR}/bud/copy-from
+
+  run_buildah bud --signature-policy ${TESTSDIR}/policy.json --jobs 4 -t ${target} -f ${TESTSDIR}/bud/copy-from/Dockerfile3 ${TESTSDIR}/bud/copy-from
+  run_buildah bud --signature-policy ${TESTSDIR}/policy.json --jobs 4 -t ${target} -f ${TESTSDIR}/bud/copy-from/Dockerfile4 ${TESTSDIR}/bud/copy-from
 }
```
Comment (TomSweeneyRedHat) [inline tests/bud.bats:1390]: The Dockerfile3 contains:

```
# cat Dockerfile3
FROM docker.io/library/busybox AS build
RUN rm -f /bin/paste
USER 1001
COPY --from=docker.io/library/busybox /bin/paste /test/
```
and Dockerfile4
```
# cat Dockerfile4
FROM docker.io/library/busybox AS test
RUN rm -f /bin/nl
FROM docker.io/library/alpine AS final
COPY --from=docker.io/library/busybox /bin/nl /test/
```

could we add some checks to make sure the images have the expected values?

File: docs/buildah-bud.md (hunk 1 above)
Comment (giuseppe) [reply docs/buildah-bud.md:283]: good idea.  Changed to accept only >= 0, with 0 unlimited

File: tests/bud.bats (hunk 2 above)
Comment (giuseppe) [reply tests/bud.bats:1404]: sure, added a check to verify the image have the same number of files

**Comment by rhatdan** [comment]
LGTM
Don't know what is wrong with the gating system?  @cevich PTAL

**Comment by giuseppe** [comment]
@rhatdan tests are passing now

**Comment by TomSweeneyRedHat** [comment]
bors retry

**Comment by giuseppe** [comment]
bors retry

**Comment by giuseppe** [comment]
bors retry

**Comment by TomSweeneyRedHat** [comment]
Giuseppe, this needs a rebase and I think that will also pull in a fix in #2427 that will help get the tests happy.

**Comment by giuseppe** [comment]
rebased and tests are green

**Comment by giuseppe** [comment]
bors retry

**Comment by rhatdan** [comment]
bors retry

**Pull request merged**